### ISSUE TITLE
no `default_environment` && `new_default_environment` in SCTK

### ISSUE THREAD

**Issue by Jedsek** [Smithay/book#18]
They were inroduced in book, but disappeared in now version.

**Comment by tsdh** [comment]
Yes, it seems the `Environment` struct with all the stuff discussed in the book were only part up to but not including smithay-client-toolkit v0.17.0 and then a big refactoring was done and a lot of what's discussed in the book vanished.

It would be very good if the book published at https://smithay.github.io/book/intro.html had a disclaimer that it corresponds to the now outdated version 0.16.0 (or whatever the right version is) and will hopefully be updated in the future (volunteers welcome!).

**Comment by Hyyperb** [comment]
Does anyone know where did these macros go I am not able to find any good resources to proceed with
